src/plugins/discover/public/context_awareness/profiles_manager.ts

```diff
@@ -208,6 +210,11 @@ export class ProfilesManager {
 
     this.ebtManager.updateProfilesContextWith(dscProfiles);
   }
+
+  private getRootRenderAppWrapper() {
+    const rootProfile = this.rootProfileService.getProfile(this.rootContext$.getValue());
+    return rootProfile.getRenderAppWrapper;
+  }
```

Comment:
The `getRenderAppWrapper` extension works a bit differently than other extensions. It can only be used at the root profile level and can only be executed once on root profile resolution. This is because we cannot change the wrapping component at runtime (e.g. when data source profiles resolve) without remounting the entire Discover application (since React cannot reparent components), causing things like loss of local state and reloading of the Lens embeddable. I believe this is an acceptable restriction for this type of extension.